dataprofiler/profilers/categorical_column_profile.py

```diff
@@ -233,10 +251,15 @@ def _update_categories(
         :type df_series: pandas.DataFrame
         :return: None
         """
-        category_count = df_series.value_counts(dropna=False).to_dict()
-        self._categories = utils.add_nested_dictionaries(
-            self._categories, category_count
-        )
+        if not self.check_stop_condition_is_met(df_series):
+            category_count = df_series.value_counts(dropna=False).to_dict()
+            # If condition for limiting profile calculations
+
+            self._categories = utils.add_nested_dictionaries(
+                self._categories, category_count
+            )
+        else:
+            self._categories = {}
```

Comment:
i think we can move the check to just the end and then do:
```python
if self.....:
    self._categories = {}
```
We are going to need to figure out how we handle the `unique_raito` calc as we probably still want to hold the last known values for profiling.